source/elements/oneDAL/source/data_management/data_sources/index.rst

```diff
@@ -0,0 +1,157 @@
+  of the numeric :capterm:`data formats <data format>`.
+
+CSV data source
+-----------------
+Class ``csv::data_source`` is an implementation of a data source type
+for which the following is true:
+
+::
+
+::
+
+   class data_source {
+   public:
+      csv_data_source(const char *file_name);
+      csv_data_source(const std::string &file_name);
+
+      char get_delimiter() const;
+      bool get_parse_header() const;
+      std::string get_file_name() const;
+
+      csv_data_source& set_delimiter(char);
+      csv_data_source& set_parse_header(bool);
+      csv_data_source& set_file_name(const char *);
+      csv_data_source& set_file_name(const std::string &);
+   };
+
+.. namespace:: onedal::csv
+.. class:: data_source
+
+   .. function:: data_source(const char *file_name)
+
+      Creates new instance of csv data source with the given :expr:`file_name`, the
+      other attributes get default values.
+
+   .. function:: data_source(const std::string &file_name)
+
+      Creates new instance of csv data source with the given :expr:`file_name`, the
+      other attributes get default values.
+
+   .. member:: char delimiter = ','
+
+      Symbol that represents delimiter between continuous feature columns in the input file.
+
+      Getter & Setter
+         | ``char get_delimter() const``
+         | ``data_source& set_delimiter(char)``
+
+   .. member:: bool parse_header = false
+
+      Flag that indicates whether or not first line in the input file should be processed
+      as a header line and should not be included in the dataset.
+
+      Getter & Setter
+         | ``bool get_parse_header() const``
+         | ``data_source& set_parse_header(bool)``
+
+   .. member:: std::string file_name = ""
+
+      String which contains name of the file with the dataset for read.
+
+      Getter & Setter
+         | ``std::string get_filename() const``
+         | ``data_source& set_file_name(const char *)``
+         | ``data_source& set_file_name(const std::string &)``
+
+-------------
+Usage example
+-------------
+::
+
+   const auto data_source = onedal::csv::data_source("data.csv")
+      .set_delimiter(',')
+
+   const auto table = ondedal::read<onedal::table>(data_source);
+
+
+Reading :expr:`onedal::read<onedal::table>(...)`
+-----------------------------------
+Args
+~~~~~
+::
+
+   template<>
+   class read_args<onedal::table> {
+   public:
+      read_args();
+
+   };
+
+.. class:: template<> read_args<onedal::table>
+
+   .. function:: read_args()
+
+      Creates arguments for the table reading operation with the default attribute
+      values.
+
+Result
+~~~~~~
+
+Result of a read operation with a specified :code:`onedal::table` template parameter is a :txtref:`table`.
+
+Operation semantics
+~~~~~~~~~~~~~~~~~~~
+.. namespace:: onedal
+.. function:: template <typename Object, typename DataSource> \
+              Object read(const DataSource& ds)
+
+   :tparam Object: oneDAL object that should be read.
```

Comment:
should be read - may not be clean. Are you trying to say this object is result of the data reading from the data source
Also, 'object' sounds a bit abstract. Need to explain details. If we have high level summary section, it will help to explain this specifics there

Reply:
Added some details.